### PR TITLE
Improve getRecords

### PR DESCRIPTION
**Pull request by dgtony**
Related to #433 

Do not request already known records for all logs contained in a thread on receiving a `PushLog` message.
It should help to reduce traffic and resources of both nodes and replicators.

### CODE REVIEW

File: net/net.go [hunk 1]
```diff
@@ -1330,25 +1306,57 @@ func (n *net) updateRecordsFromLog(tid thread.ID, lid peer.ID) {
...
 	}
 
-	// Get log records for this new log
-	recs, err := n.server.getRecords(
-		n.ctx,
-		tid,
-		lid,
-		map[peer.ID]cid.Cid{lid: cid.Undef},
-		MaxPullLimit)
+	// TODO after protocol change request only new log
```
Comment (sanderpick) [inline net/net.go:1309]: So, are you wanting to follow this up with a protocol change ASAP?

File: net/record.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+			return
+		}
+	}
+
+	// start a new fragment
+	s.fragments = append(s.fragments, []linkedRecord{rec})
+}
+
+// return reconstructed sequence and success flag
+func (s *recordSequence) List() ([]linkedRecord, bool) {
```
Comment (sanderpick) [inline net/record.go:95]: Very nice, clever 💯

**Review by sanderpick — APPROVED** [review]
LGTM! One question regarding protocol change procedure.

File: net/net.go (hunk 1 above)
Comment (dgtony) [reply net/net.go:1309]: Well, as we previously discussed it, protocol change requires getting rid of record integrity check, and you've suggested that the next version of threads protocol should be issued. It'd be great, but not urgent now, as current PR already significantly reduces number of irrelevant records returned by the GetRecords request.

Comment (sanderpick) [reply net/net.go:1334]: Okay, SGTM. I have some other changes that will be nice to get in before a protocol change. So, will be nice to wait for those.

**Pull request merged**